linera-core/src/worker.rs

```diff
@@ -708,6 +704,45 @@ where
         Ok(sender)
     }
 
+    /// Retrieves an endpoint to a [`ChainWorkerActor`] from the cache, attempting to create one
+    /// and add it to the cache if needed.
+    ///
+    /// Returns [`None`] if the cache is full and candidate for eviction was found.
```

Comment:
Do you mean "no candidate"?

I didn't realize this is introducing a new failure scenario by the way. We should probably use a cap as high as possible by default then (like half of the tokio spawn_blocking pool)

Reply:
Yes, typo fixed.

As we discussed, the cap for validators is 400, which should be enough for now.